### PR TITLE
MLMediaFinder implementation

### PR DESCRIPTION
**Pull request by munxar**

### CODE REVIEW

File: formwidgets/MLMediaFinder.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        // make root path of media files accessible
+        $this->vars['mediaPath'] = $this->mediaPath = MediaLibrary::url('/');
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function loadAssets()
+    {
+        // we only load the css from parent, unfortunately the js must be reimplemented due to missing events.
```
Comment (LukeTowers) [inline formwidgets/MLMediaFinder.php]: I don't want to merge this with copied JS, propose a PR to the October mediafinder JS to add any events that would be required to accomplish your goal here.

Comment (scorewinner) [reply formwidgets/MLMediaFinder.php]: Hey, @LukeTowers we changed the code, it is modular now and can be merged. 
It needs the change in the mediafinder https://github.com/octobercms/october/pull/3446 in order to work.

**Comment by LukeTowers** [comment]
@munxar can you update this to work with https://github.com/rainlab/translate-plugin/pull/416 and update the version.yaml to fix the conflicts?

**Comment by munxar** [comment]
@LukeTowers I'd love too, but first you'll have to merge this PR:
https://github.com/octobercms/october/pull/3446/files

**Comment by LukeTowers** [comment]
@munxar done

**Comment by munxar** [comment]
@LukeTowers this PR is now up to date and ready to be merged.

**Comment by LukeTowers** [comment]
@munxar could you submit a GIF showing how this field works?

**Comment by munxar** [comment]
@LukeTowers here you go:
mlmediafinder

File: formwidgets/MLMediaFinder.php [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+{
+    use \RainLab\Translate\Traits\MLControl;
+
+    /**
+     * @inheritDoc
+     */
+    protected $defaultAlias = 'mlmediafinder';
+
+    public $originalAssetPath;
+    public $originalViewPath;
```
Comment (LukeTowers) [inline formwidgets/MLMediaFinder.php]: Why are these still here?

File: formwidgets/MLMediaFinder.php [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    private $mediaPath;
```
Comment (LukeTowers) [inline formwidgets/MLMediaFinder.php]: Could you add a docblock comment for this property?

File: formwidgets/MLMediaFinder.php [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    {
+        return base_path().'/modules/backend/formwidgets/mediafinder/partials';
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function getParentAssetPath()
+    {
+        return '/modules/backend/formwidgets/mediafinder/assets';
```
[the remark this answers is not in the document]
Comment (munxar) [reply formwidgets/MLMediaFinder.php]: I'll just do what all other controls do. Can't we do a refactoring of the MLControl trait in another PR please?

[the remark this answers is not in the document]
Comment (LukeTowers) [reply formwidgets/MLMediaFinder.php]: Fair enough. Sounds good to me

**Pull request merged**